powerbi-docs/developer/visuals/license-api.md

```diff
@@ -0,0 +1,188 @@
+
+Example of calling `getAvailableServicePlans`:  
+
+```typescript
+this.licenseManager.getAvailableServicePlans().then((result: LicenseInfoResult) => {
+            this.notificationType = result.isLicenseUnsupportedEnv ?  powerbi.LicenseNotificationType.UnsupportedEnv : powerbi.LicenseNotificationType.General;
+            this.hasServicePlans = !!(result.plans && result.plans.length && result.plans[0].spIdentifier == "myServiceId" && 
+                ( result.plans[0].state == powerbi.ServicePlanState.Active ||  result.plans[0].state == powerbi.ServicePlanState.Warning));
+            
+            // display notification if the user doesn't have licenses
+            if (!this.hasServicePlans) {
+                this.licenseManager.notifyLicenseRequired(this.notificationType).then((value) => {
+                    if (value) {
+                        this.isIconDisplayed = true;
+                    }
+                }).catch((err) => {
+                    console.log('ERROR', err);
+                })
+            }
+        }).catch((err) => {
+            this.hasServicePlans = undefined;
+            console.log(err);
+        });
+```
+
+## Notify the user that the required licenses are missing
+
+Power BI platform provides several out-of-the-box experiences that can be used to notify the user that:
+
+* Licenses should be purchased in order to enjoy the visual's full capabilities
+* A particular visual's feature is blocked due to missing licenses
+* Entire visual is blocked due to missing licenses
+* Entire visual is blocked because the Power BI environment used doesn't support license management\enforcement
+
+```typescript
+export interface IVisualLicenseManager {
+        notifyLicenseRequired(notificationType: LicenseNotificationType): IPromise<boolean>;
+        notifyFeatureBlocked(tooltip: string): IPromise<boolean>;
+        clearLicenseNotification(): IPromise<boolean>;
+    }
+```
+
+
+### General icon to indicate that the required licenses are missing
+
+Use `notifyLicenseRequired` call with `LicenseNotificationType.General` to display an icon as part of the visual's container.  
+Once triggered, the icon will be preserved throughout the visual's lifetime until until `clearLicenseNotification` or `notifyLicenseRequired` are called..
+
+> [!NOTE]
+> The `LicenseNotificationType.General` notification is only enforced when it is called in a supported licensing environment **and** Power BI is in Edit mode. Calling this in an unsupported environment or when the report is in Read mode or in dashboard will return `false` in the call's response and the icon won't appear.    
+
+Example of the visual display containing the "licenses are required" general icon:
+
+>[!div class="mx-imgBorder"]
+>![visual display containing the "licenses are required" general icon.](media/licensing/generalIcon.png)
```

Comment:
Images should be saved in a folder with the same name as the file. In this case, license-api